src/API/Google/Merchant.php

```diff
@@ -316,6 +316,10 @@ public function link_ads_id( int $ads_id ): bool {
 		$account   = $this->get_account();
 		$ads_links = $account->getAdsLinks();
 
+		if ( ! is_array( $ads_links ) ) {
```

Comment:
@mikkamp while working on this feature, we ran into a bug here because it looks like there are cases where the `Google\Service\ShoppingContent\Account` object returned when you run `$this->get_accounts()` above can contain a `null` value for the `$adsLinks` property, which then throws an error below. 

This may have happened due to the fact that I've created multiple MC accounts under the same domain while testing. Either way, adding this bit of code fixes the issue here, but unsure if a more general fix is needed. Let me know if you have any questions or need more info.

Reply:
Great catch @joemcgill although would you be able to provide a little more detail on how you are getting this issue. I'm not able to reproduce this locally, both with a MC account that never linked an Ads account and an account where previous links have been removed/cancelled I'm still unable to get it to return null from `getAdsLinks`.

What's happening here is I'm getting a REST response without the `adsLinks` field included, since the Account object is based off a [Model](https://github.com/googleapis/google-api-php-client/blob/v2.15.3/src/Model.php), when it calls mapTypes it doesn't set the `adsLinks` property (mapTypes is called from the constructor when the [REST class creates the object](https://github.com/googleapis/google-api-php-client/blob/v2.15.3/src/Http/REST.php#L144)). So when it calls `getAdsLinks` it calls the magic `__get` function from the Model class. Which as we [can see here](https://github.com/googleapis/google-api-php-client/blob/v2.15.3/src/Model.php#L70) will return an empty array based on the property `adsLinksDataType`.

It's no harm to have another check to set it to an empty array, but I'm just curious why that wouldn't have been reported before, as we have plenty of MC accounts linked to an Ads account with all kinds of configurations.

Reply:
Also I saw the latest code was changed to doing a typecast:
```php
$ads_links = (array) $account->getAdsLinks();
```

If we are going to expect the function to return `null` in some cases then IMHO it would be better to use the null coalescing operator so if it does return an unexpected value it can catch and log it as an exception.
```diff
- $ads_links = (array) $account->getAdsLinks();
+ $ads_links = $account->getAdsLinks() ?? [];
```